lib/puma/const.rb

```diff
@@ -147,7 +147,7 @@ module Const
 
     REQUEST_METHOD = "REQUEST_METHOD"
     HEAD = "HEAD"
-    SUPPORTED_HTTP_METHODS = %w[HEAD GET POST PUT DELETE OPTIONS TRACE PATCH].freeze
+    DEFAULT_SUPPORTED_HTTP_METHODS = %w[HEAD GET POST PUT DELETE OPTIONS TRACE PATCH].zip([nil]).to_h.freeze
```

Comment:
I think we should add a comment explaining why we do `.zip([nil]).to_h` here, it is not obvious 

I'll include what this is evaluated to for reference 

```ruby
irb(main):001:0> %w[HEAD GET POST PUT DELETE OPTIONS TRACE PATCH].zip([nil]).to_h
=> {"HEAD"=>nil, "GET"=>nil, "POST"=>nil, "PUT"=>nil, "DELETE"=>nil, "OPTIONS"=>nil, "TRACE"=>nil, "PATCH"=>nil}
```

Reply:
Also, should we do this? The previous implementation was fine with a regular Array? Do we know if it matters?